src/peep/peep.c

```diff
@@ -7772,13 +7772,370 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
 	}
 }
 
-/**
+/** Main logic to decide whether a peep should buy an item in question
+ *
+ * Seems to also handle the actual purchase as well, so once it returns, the
+ * peep will have the item and the money will have been deducted.
+ *
+ * eax: shopItem | (rideIndex << 8)
+ * ecx: price
+ * esi: peep*
+ *
+ * Returns 0 or 1 depending on if the peep decided to buy the item
  *
  *  rct2: 0x0069AF1E
  */
 static bool sub_69AF1E(rct_peep *peep, int rideIndex, int shopItem, money32 price)
 {
-	return !(RCT2_CALLPROC_X(0x0069AF1E, shopItem | (rideIndex << 8), 0, price, 0, (int)peep, 0, 0) & 0x100);
+	rct_ride* ride = GET_RIDE(rideIndex);
+	sint16 value;
+	int shopItem2 = (shopItem >= 32 ? (shopItem - 32) : shopItem); // A separate item that's used as output
+
+	///sub_69AF1E
+	bool has_voucher = false;
+
+	if ((peep->item_standard_flags & PEEP_ITEM_VOUCHER) && // If they have a voucher...
+		(peep->voucher_type == VOUCHER_TYPE_FOOD_OR_DRINK_FREE) && // ...and it's a free food/drink voucher...
+		(peep->voucher_arguments == shopItem)) { // ...and it matches the item in question?...
+		has_voucher = true;
+	}
+
+///loc_69AF47:
+	// Check to see if the peep already has the item that they're considering buying
+	if (peep_has_item(peep, shopItem)) {
+		peep_insert_new_thought(peep, PEEP_THOUGHT_TYPE_ALREADY_GOT, shopItem);
+		return 0;
+	}
+
+///loc_69AF6B:
+	// Checks to see if the peep is already eating/drinking or is too nauseous
+	// TODO: check to make sure all of the conditions are covered
+	if (shop_item_is_food_or_drink(shopItem)) {
```

Comment:
It's not. The bit test is done on 0x36BA3E0 (standard) and 0x11FF78 (extra). The standard flag matches 1 for any food or drink item in the peep item enums. The same masks are used to check the peep's items.